src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @SuppressWarnings("unchecked")
+    @Override
+    public List<T> getAll(Class<? extends Machine> type) {
```

Comment:
Consider creating a local variable for the result of the 'get()' call to avoid casting the list multiple times. This would improve readability and maintainability of the code.